src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentContext.cs

```diff
@@ -37,9 +40,10 @@ internal class DocumentContext
 
     public virtual IProjectSnapshot Project => Snapshot.Project;
 
-    public virtual TextDocumentIdentifier Identifier => new TextDocumentIdentifier()
+    public virtual TextDocumentIdentifier Identifier => new VSTextDocumentIdentifier()
```

Comment:
I wonder if it makes more sense to change the `Uri` parameter to a `TextDocumentIdentifier` parameter, that way it would carry the project context with it, without needing to change every constructor call etc. Would be more future proof too, if anything else is added to VSTextDocumentIdentifier.

Could also mean we can then have an overload of `DocumentContextFactory.TryCreate()` that took a `TextDocumentIdentifier`, and one that took a `Uri` and assumed null for the project context, which would clean up a lot of the explicit null passing you've had to do, but still easy to find code that should be updated to add in a context, but doing a FAR on that overload.

Reply:
That's a good point. I like the new shape a bit better. See latest commit

Reply:
I was thinking the `DocumentContext` itself could store the `TextDocumentIdentifier`, but I see now that wouldn't work if using the `TryCreate` method that takes a `Uri`. Are there cases where we have a `ProjectContext`, but also only have a `Uri`?

This is certainly a nicer shape, but still has the potential for data loss in future.

Reply:
Update: Yes, we do have such an overload. hmm. 

Here is my thinking: If we had always fully preserved the TextDocumentIdentifier, we wouldn't need this PR at all (in theory) as everything would just flow. By constructing this `new VSTextDocumentIdentifier` here, we are explicitly breaking the flow of the underlying Json data, but I don't have a strong sense of how likely such a break will be in future.